File: app/src/main/java/com/zhufuc/pctope/Utils/DiffUtilCallback.java
package com.zhufuc.pctope.Utils;

import android.support.v7.util.DiffUtil;

import com.zhufuc.pctope.Adapters.TextureItems;

import java.util.Objects;

/**
 * Created by zhufu on 17-7-29.
 */

public class DiffUtilCallback extends DiffUtil.Callback {

    private TextureItems oldTemp,newTemp;

    public DiffUtilCallback(TextureItems oldTemp,TextureItems newTemp){
        this.oldTemp = oldTemp;
        this.newTemp = newTemp;
    }

    @Override
    public int getOldListSize() {
        return oldTemp.getItemCount();
    }

    @Override
    public int getNewListSize() {
        return newTemp.getItemCount();
    }

    @Override
    public boolean areItemsTheSame(int oldItemPosition, int newItemPosition) {
        return Objects.equals(oldTemp.getItem(oldItemPosition),newTemp.getItem(newItemPosition));
    }

    @Override
    public boolean areContentsTheSame(int oldItemPosition, int newItemPosition) {
        return Objects.equals(oldTemp.getItem(oldItemPosition).getName(),newTemp.getItem(newItemPosition).getName());
    }
}

File: app/src/main/java/com/zhufuc/pctope/Utils/DiffUtilCallback.kt
package com.zhufuc.pctope.Utils

import android.support.v7.util.DiffUtil
import android.support.v7.widget.RecyclerView

import com.zhufuc.pctope.Adapters.TextureItems

import java.util.Objects

/**
 * Created by zhufu on 17-7-29.
 */

class DiffUtilCallback(private val oldTemp: RecyclerView.Adapter<*>, private val newTemp: RecyclerView.Adapter<*>) : DiffUtil.Callback() {

    override fun getOldListSize(): Int {
        return oldTemp.itemCount
    }

    override fun getNewListSize(): Int {
        return newTemp.itemCount
    }

    override fun areItemsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldTemp.getItemId(oldItemPosition) == newTemp.getItemId(newItemPosition)
    }

    override fun areContentsTheSame(oldItemPosition: Int, newItemPosition: Int): Boolean {
        return oldTemp.javaClass == newTemp.javaClass
    }
}
